packages/headless/src/modules/global-search/services/elastic.service.ts

```diff
@@ -0,0 +1,178 @@
+      let data: any = {} 
+
+      data.infoProductId = e.id || ''
+      data.infoName = e?.info?.name || ''
+      data.infoShortDescription = e?.info?.shortDescription || ''
+      data.infoFullDescription = e?.info?.fullDescription || ''
+      data.infoSku = e?.info?.sku || ''
+      data.infoPrice = e?.info?.price || 0
+
+      data.metaKeywords = e?.meta?.keywords || []
+      data.metaTitle = e?.meta?.title || ''
+      data.metaDescription = e?.meta?.description || ''
+
+      data.brands = e?.brands || []
+      data.categories = e?.categories || []
+      data.manufacturer = e?.manufacturer  || {}
+      data.photos = e?.photos.map(p => p.url)  || []
+      data.tags = e?.tags || []
+
+      return data;
+  } 
+
+  async getSingleByElasticId(id: string): Promise<any>{
+    const {body} =  await this.esService.get({index: searchConfig.index, id})
+    return body._source
+  }
+
+  async singleInsert(productData: Product): Promise<number> { 
+    try { 
+      const index = searchConfig.index 
+      // if not exist
+      const body = this.mapSearchData(productData) 
+      const result = await this.esService.index({index,refresh: true, body}) 
+      return result.statusCode;
+    } catch (error) {
+      console.log(error)
+      return 0;
+    } 
+  }
+
+async deleteSingleByElasticId(id: string): Promise<number> { 
+    try { 
+      const index = searchConfig.index
+      const type = searchConfig.index 
+      const result = await this.esService.delete({ index, type,  id }) 
+      return result.statusCode;
+    } catch (error) {
+      console.log(error)
+      return 0;
+    } 
+  }
+
+  async bulkInsert(){
+      try {
+      await this.createIndex();
+      await this.createSchema();
+
+      const allProduct =  await this.productSearchDB.findAllProducts()
+      if(!allProduct || allProduct.length < 1){
+          return 0;
+      }
+      const esAction = {
+        index: {
+          _index: searchConfig.index,
+          _type: searchConfig.index
+        }
+      } 
+      let docs = []  
+      const mapped = allProduct.map(e =>{ 
+          const data = this.mapSearchData(e)  
+          docs.push(esAction); 
+          docs.push(data); 
+      })   
+      const result = await this.esService.bulk({ body: docs }) 
+      await this.esService.indices.refresh({index: searchConfig.index})
+      return result.statusCode; 
+      } catch (error) {
+      console.log(error)
+      }
+  }
+
+
+  /**
+   * global product search - public api 
+   */
+  async search(keyword: string): Promise<IServiceResponse<ISearchProductResponse>>{
+    const result = await this.getSearchData(keyword)
+    if(!result){
+      return errorResponse("Error while searching", null, HttpStatus.CONFLICT)
+    }
+    return successResponse(null, result, HttpStatus.OK)
+  }
+ 
+  private async getSearchData(searchKey): Promise<ISearchProductResponse> { 
+    const query = {
+      from: 0,
+      size: 4,
```

Comment:
update these value. take it from request

Reply:
done